assets/js/modules/analytics-4/datastore/conversion-reporting.test.js

```diff
@@ -37,6 +38,32 @@ describe( 'modules/analytics-4 conversion-reporting', () => {
 	} );
 
 	describe( 'actions', () => {
+		describe( 'receiveConversionReportingInlineData', () => {
+			it( 'requires the data param', () => {
+				expect( () => {
+					registry
+						.dispatch( MODULES_ANALYTICS_4 )
+						.receiveConversionReportingInlineData();
+				} ).toThrow( 'data is required.' );
+			} );
+
+			it( 'receives and sets inline data', async () => {
+				const data = {
+					newEvents: [ 'purchase' ],
+					lostEvents: [],
+				};
+
+				await registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.receiveConversionReportingInlineData( data );
+
+				expect(
+					registry
+						.select( MODULES_ANALYTICS_4 )
+						.getConversionReportingEventsChange()
```

Comment:
Actions shouldn't use selectors in their tests, but rather make assertions on state directly. Action tests test the action creator + reducer, selector tests test the state, resolver + selector. e.g. https://github.com/google/site-kit-wp/blob/2610734a9dc12c7964a85f8c6c4b1879d43fb104/assets/js/modules/analytics-4/datastore/properties.test.js#L200-L205

Reply:
Thanks, updated